app/assets/javascripts/controllers/schedules/controller.js

```diff
@@ -8,6 +8,44 @@
 Yacs.views.schedules = function (target, params) {
   Yacs.render(target, 'schedules');
 
+  var schedule_ids = [];
+
+  // check for query parameters
+  var query = window.location.hash.split('?')[1]
+  if(typeof query !== 'string' || query.length < 1) {
+    /* If there are no query parameters, make the current
+     * selected section ids into a query parameter and
+     * re-navigate the page to that.
+     * This is so someone who clicks on the Schedule button
+     * after selecting courses will get the copiable link in
+     * their address bar.
+     */
+    Yacs.router.visit('/schedules?section_ids='+ Yacs.user.getSelectionsRaw());
```

Comment:
Rather than always have the ids in the url, could we have it use the cookie if there are no ids present in the query and use the query if it is present? The long query looks ugly and i'd rather only have it there when necessary.